scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -117,23 +117,36 @@ object ScioContext {
  * @groupname Ungrouped Other Members
  */
 // scalastyle:off number.of.methods
-class ScioContext private[scio] (val options: DataflowPipelineOptions,
+class ScioContext private[scio] (val options: PipelineOptions,
                                  private var artifacts: List[String],
                                  testId: Option[String]) {
 
   private val logger = LoggerFactory.getLogger(ScioContext.getClass)
 
   import Implicits._
 
+  // TODO: decouple DataflowPipelineOptions
+  private def dfOptions: DataflowPipelineOptions = try {
+    options.as(classOf[DataflowPipelineOptions])
+  } catch {
+    case _: Throwable =>
+      val name = options.getClass.getSimpleName
+      throw new RuntimeException(s"$name is not DataflowPipelineOptions")
+  }
+
   // Set default name if no app name specified by user
-  if (options.getAppName == null || options.getAppName.startsWith("ScioContext$")) {
-    this.setName(CallSites.getAppName)
+  if (options.isInstanceOf[DataflowPipelineOptions]) {
+    if (dfOptions.getAppName == null || dfOptions.getAppName.startsWith("ScioContext$")) {
+      this.setName(CallSites.getAppName)
+    }
   }
 
   /** Dataflow pipeline. */
   def pipeline: Pipeline = {
     if (_pipeline == null) {
-      options.setFilesToStage(getFilesToStage(artifacts).asJava)
```

Comment:
I just don't like the silent failure - so if someone adds artifacts, and if there is more runners some time from now, will someone remember to look in this piece of code to make sure artifacts are handled.